modules/compute/virtual-machine/main.bicep

```diff
@@ -567,19 +567,21 @@ resource vm_logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021
   scope: az.resourceGroup(split(monitoringWorkspaceId, '/')[2], split(monitoringWorkspaceId, '/')[4])
 }
 
-module vm_microsoftMonitoringAgentExtension 'extension/main.bicep' = if (extensionMonitoringAgentConfig.enabled) {
-  name: '${uniqueString(deployment().name, location)}-VM-MicrosoftMonitoringAgent'
+module vm_azureMonitorAgentExtension 'extension/main.bicep' = if (extensionMonitoringAgentConfig.enabled) {
```

Comment:
As per similar discussion for VMSS (ref https://github.com/Azure/ResourceModules/pull/3918#discussion_r1316017695) we won't be including this change into release 0.11 to avoid an additional breaking change. We should discuss and decide on if keeping both AMA and MMA extensions or if discontinuing support for MMA

Reply:
Same as for the other, I believe this can and should be merged

Reply:
ping @eriqua :D By now already have quite some conflicts

Reply:
Plus I believe @rahalan is already working on the migration into AVM. So it's high time this is merged.

Reply:
ping @eriqua

Reply:
Looping in @rahalan, AVM VM and VMSS module owner, to finalize the decision on if keeping the legacy MMA support or just remove it as part of migration to AVM.

Reply:
Has to be merged into AVM module as soon as its [PR](https://github.com/Azure/bicep-registry-modules/pull/717) is merged